src/orion/algo/tpe.py

```diff
@@ -594,25 +663,26 @@ def __init__(
         self.attempts_factor = attempts_factor
         self.max_attempts = max_attempts
 
-        self.pdfs = []
+        self.pdfs: list[rv_frozen] = []
         self._build_mixture()
 
-    def _build_mixture(self):
+    def _build_mixture(self) -> None:
         """Build the Gaussian components in the GMM"""
         for mu, sigma in zip(self.mus, self.sigmas):
             self.pdfs.append(norm(mu, sigma))
 
-    def sample(self, num=1, attempts=None):
+    def sample(self, num: int = 1, attempts: int | None = None) -> list[numpy.ndarray]:
         """Sample required number of points"""
         if attempts is None:
             attempts = self.base_attempts
-
-        point = []
+        # TODO: See below, unsure about the return type for this function.
```

Comment:
Seems like numpy.ndarray or scalar? https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_continuous.rvs.html

Reply:
Fixed in 040db502d7c79823ef8130affb10a76f911198a1